src/hdr.imageio/rgbe.cpp

```diff
@@ -106,6 +106,28 @@ float2rgbe(unsigned char rgbe[4], float red, float green, float blue)
   }
 }
 
+
+inline void
+float2rgbe(unsigned char* rgbe, const float* rgb)
+{
+    float red = rgb[0], green = rgb[1], blue = rgb[2];
+    float v = red;
+    if (green > v) v = green;
+    if (blue > v) v = blue;
+    if (v < 1e-32) {
+        rgbe[0] = rgbe[1] = rgbe[2] = rgbe[3] = 0;
+    } else {
+        int e;
+        v = frexpf(v,&e) * 256.0f/v;
```

Comment:
Not related to this review, but these encoding/decoding method cannot be optimized by the compiler (just checked gcc/clang in godbolt).

Its fairly easy to swap in some bit-twiddling in its place.

Reply:
Certainly true. It's just a matter of how much effort to put into a format that is never really on a critical performance path.